### DISCUSSION TITLE
Resent-Sender

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2396] [Help with using Postal]
I am trying to ensure our mails dont end up in spam. I am getting far, but it seems that some providers check the DKIM of the return path email (Resent-Sender). Is there a way to fix this?

Is this header needed? I cannot find this header when using other transactional mail providers 

I have tried setting it to false in my config, but it does not seem to make any difference (it is still added)
https://github.com/postalserver/postal/blob/main/config/postal.defaults.yml#L20

Edit: Ah this might not be released yet? Just noticed that the PR was from last week

**Comment by sinnbeck** [comment]
An example of a failed DMARC check
```
A message claiming to be from you has failed the published DMARC policy for your domain.

  Sender Domain: mail.aalund.com
  Sender IP Address: 88.99.190.207
  Received date: Wed, 29 Mar 2023 15:54:46 +0200
  SPF Alignment: no
  DKIM Alignment: no
  DMARC Results:

------ This is a copy of the headers that were received before the error was detected.

Received: from postal.aalund.com ([88.99.190.207])
        by node8.spamfilter.io with esmtps (TLSv1.3:TLS_AES_256_GCM_SHA384:256)
        (Exim 4.92)
        (envelope-from <mremqm@psrp.mail.aalund.com>)
        id 1phWGE-000HgP-07
        for info@[redacted]; Wed, 29 Mar 2023 15:54:46 +0200
Resent-Sender: mremqm@psrp.mail.aalund.com
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=mail.aalund.com;
        s=postal-UaLZbl; t=1680097450;
        bh=ruTrqkKx0PZDl5kGQlYUmsm1Xn4bgO7GAF31HQuWERg=;
        h=from:to:reply-to:subject:message-id:mime-version:date:content-type;
        b=oA+2VeN2yRuW5iSHAQ2w94ivB1GFW7e4RcptvI2AQYatar2Bxg5dU2vZKPjabEygVASBarRy
        mH6wQQEr13FaSStPRzkojC/fU2LXHUkeNChYEPK0YwckElH61ex7ylTuWQDcP/3VWqrGDHu9
        vg0gC6j6Fv2W8wdQQLQPXUf6lIE=
X-Postal-MsgID: NsECpQXJmzYc
Received: from [127.0.0.1] (2a01:4f8:1c1e:70b6::1 [2a01:4f8:1c1e:70b6::1]) by postal.aalund.com with SMTP; Wed, 29 Mar 2023 13:44:10 -0000
From: Info <info@mail.aalund.com>
To: [redacted]
Subject: [redacted]
Message-ID: <c7dc90584654f5a033868538f6283359@mail.aalund.com>
MIME-Version: 1.0
Date: Wed, 29 Mar 2023 15:44:10 +0200
Content-Type: multipart/alternative; boundary=8cHRWWwr
Received-SPF: pass (node8.spamfilter.io: domain of psrp.mail.aalund.com designates 88.99.190.207 as permitted sender) client-ip=88.99.190.207; envelope-from=mremqm@psrp.mail.aalund.com; helo=postal.aalund.com;
X-SPF-Result: node8.spamfilter.io: domain of psrp.mail.aalund.com designates 88.99.190.207 as permitted sender
X-DKIM-Status: none /  / psrp.mail.aalund.com /  /  /
X-DKIM-Status: pass /  / mail.aalund.com / mail.aalund.com /  / postal-UaLZbl
Authentication-Results: node8.spamfilter.io;
        iprev=pass (postal.aalund.com) smtp.remote-ip=88.99.190.207;
        spf=pass smtp.mailfrom=psrp.mail.aalund.com;
        dkim=pass header.d=mail.aalund.com header.s=postal-UaLZbl header.a=rsa-sha256;
        dmarc=skipped
Authentication-Results:  spamfilter.io; spf=pass smtp.mailfrom=mremqm@psrp.mail.aalund.com; dkim=pass header.i=mail.aalund.com

**Reply by willpower232** [reply]
```
X-DKIM-Status: none /  / psrp.mail.aalund.com /  /  /
```

Have you added `postal._domainkey.rp.postal.example.com` as per https://docs.postalserver.io/install/dns-configuration#return-path?

**Reply by sinnbeck** [reply]
Yeah that is the strange part. https://mxtoolbox.com/SuperTool.aspx?action=dkim%3apostal._domainkey.rp.postal.aalund.com&run=networktools

And as I read the header itself, it says it everything else is passing. I am really confused :)

**Reply by willpower232** [reply]
I presume that the issue might be you're sending email from the domain you're using to identify your postal server so potentially something is getting confused because it is hard to tell what should be happening.

You could confirm this by setting up a completely different domain on your Postal server and sending from that to see what happens.

**Reply by sinnbeck** [reply]
I am sending from a different domain. mal.aalund.com is the sender domain and postal.aalund.com is the mailserver. I am not sure I understand why that would fail.

**Reply by willpower232** [reply]
ah my mistake, yeah I can't see anything wrong or at least different with what we have.

I'm not sure if turning off the resent-sender will be enough as it is also listed in the "envelope from" apparently but this is not my area of expertise so I might be mistaken.

You could try applying the patch yourself in your own copy if you can't wait for them to finish the next version, apparently the "latest" docker image tag is a bit broken so might not be great to switch to just yet.

**Reply by sinnbeck** [reply]
I have signed up for postmarks weekly mail for DMARK and will leave it on for a week, and then see how their report is looking. Hopefully it gives 100% alignment for both.
